src/util/string.h

```diff
@@ -58,34 +58,29 @@ void ReplaceAll(std::string& in_out, const std::string& search, const std::strin
 }
 
 /**
- * Join a list of items
+ * Join items
```

Comment:
That'd be incorrect. It isn't the type of the first item, but the type returned by the unary op executed on the first item. In practise this will (almost) always be `std::string`, so it seems odd to be overly verbose?

Reply:
Okay, no strong feelings about it. For a helper function that's meant to be used all over the codebase I just think it's worth thinking more about a helpful docstring, maybe a join function is not be trivial to everyone? Final suggestion but I don't want to waste time over this so feel free to ignore if you don't think it's helpful:
```suggestion
 * Join all container items. Typically used to concatenate strings but accepts containers with
 * elements of any type.
```

Reply:
thx, done